### ISSUE TITLE
StringList parameter for plugins

### ISSUE THREAD

**Issue by NicolasRannou** [FNNDSC/ChRIS_ultron_backEnd#24]
Jorge Bernal Oh, If I understood your question then yes there is a TYPES list in `models.py` in the plugins Django app

[5:40 PM]  
Nicolas Rannou yes that was that!

[5:40 PM]  
Jorge Bernal The TYPES dictionary has been used in other modules like `manager.py` for example (edited)

[5:41 PM]  
Nicolas Rannou OK

[5:41 PM]  
do we also support “list” of string for instance?

[5:42 PM]  
the use case is for a tractography plugin for instance, user can choose the mode of reconstruction he wants to run

[5:44 PM]  
Jorge Bernal Well i f you look at `models.py` so far we only support these parameter types in the DB: `StringParameter`, `IntParameter`, `FloatParameter` and `BoolParameter`

[5:44 PM]  
Nicolas Rannou yes - I was wondering maybe more types were specified somewhere else

[5:47 PM]  
Jorge Bernal If we want to implement a drop down-like list we probably will need to define another parameter type, something like `StringListParameter` which basically uses a dictionary similarly to  `TYPE_CHOICES` in the same way as the Plugin model uses `TYPE_CHOICES`

[5:48 PM]  
Another example is `PluginParameter` model using the dictionary `PLUGIN_TYPE_CHOICES` (edited)

[5:50 PM]  
So basically you have to define the possible choices in a dictionary `CHOICES` ant then in the model doing including something like `choices=CHOICES`

[5:52 PM]  
So probably the way of implementing this is a bit more complicated we can have another model like this:
```class StringChoiceParameter(models.Model):
    value = models.CharField(max_length=200, blank=True)
    plugin_inst = models.ForeignKey(PluginInstance, on_delete=models.CASCADE,
                                    related_name='string_param')
    plugin_param = models.ForeignKey(PluginParameter, on_delete=models.CASCADE,
                                     related_name='string_inst')

    def __str__(self):
        return self.value
